### PR TITLE
[51170] Gantt: shorten menu and add "Show relations" action

### PR DESCRIPTION
**Pull request by bsatarnejad**
- [x] Change menu actions on Gantt view
- [x] Add a new action entry for showing relations of a WP
- [x] Write a test for showing relations in Gantt view

https://community.openproject.org/work_packages/51170/activity

### CODE REVIEW

File: frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts [hunk 1]
```diff
@@ -88,6 +90,10 @@ export class WorkPackageContextMenuHelperService {
 
     allowedActions = allowedActions.concat(this.getAllowedParentActions(workPackage));
 
+    // remove some actions on Gantt
+    if (!!workPackage.addRelation && this.wpViewTimeline.isVisible) {
```
Comment (HDinger) [inline frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts]: This checks for permissions to add relations, but we wan't this special menu always when the timeline is visible. I'd also vote for renaming the variable to something more gantt specific.

File: frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts [hunk 2]
```diff
@@ -212,6 +218,11 @@ export class WorkPackageContextMenuHelperService {
         text: I18n.t('js.relation_buttons.add_follower'),
         link: 'addRelation',
       });
+      allowedActions.push({
+        key: 'relations',
+        text: I18n.t('js.relation_buttons.show_relations'),
+        link: 'relations',
+      });
```
Comment (HDinger) [inline frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts:230]: Same as above, this should not only be visible when the user has the permission to add relations, but always when the Gantt is visible.

**Review by HDinger — CHANGES_REQUESTED** [review]
Hi @bsatarnejad 

I tried to write down what we just discussed.

* Taken from the ticket description, there are two menu entries missing on the Gantt `Copy link to clipboard` and `Indent hierarchy`

**Comment by ulferts** [comment]
@bsatarnejad , @HDinger , the failing spec should disappear when merging with dev

**Review by HDinger — COMMENTED** [review]
The changes look good to me 👍  When testing it, I noticed that "Log unit costs" is still shown in the menu. In the ticket it is not specified but I assume that it should be hidden as "log time" is also hidden. After that, it is good to merge 👍

**Review by HDinger — APPROVED** [review]

**Pull request merged**